nrclient/nrclient.go

```diff
@@ -63,7 +63,7 @@ func (nrClient *NRClient) Send(logRecords []record.LogRecord) (retry bool, err e
 
 		// ...unless we receive an explicit non-2XX HTTP status code from the server that tells us otherwise
 		if statusCode/100 != 2 && isStatusCodeRetryable(statusCode) {
-			return true, nil
+			return true, fmt.Errorf("received non-2XX HTTP status code: %d", statusCode)
 		}
 	}
 	return false, nil
```

Comment:
So if we get a non-2xx code that _isn't_ retryable, we don't log the status code? 🤔 I wonder (a) if it's also important to log non-retryable codes as well, but (b) if that will cause too much logging, dunno.

Do you have an example of what this will look like if a customer gets 429s or something?

Reply:
This is a recent example for a 429:
```
time="2023-03-29T15:01:58Z" level=info msg="Retryable error received. Will retry to send the logs (if there are attempts remaining, check Retry_Limit option)" error="<nil>"
```

It'll look like this after upgrading:
```
time="2023-03-29T15:01:58Z" level=info msg="Retryable error received. Will retry to send the logs (if there are attempts remaining, check Retry_Limit option)" error="received non-2XX HTTP status code: 429"
```

Reply:
And you're right... the output for Non retriable errors is not including the status code neither, and IMHO it should. Added on c21d354